pyvera/subscribe.py

```diff
@@ -123,23 +123,33 @@ def _run_poll_server(self):
                 device_data, timestamp = (
                     controller.get_changed_devices(timestamp))
                 LOG.debug("Poll returned")
+
                 if self._exiting:
                     continue
-                if not device_data:
+
+                if device_data:
+                    self._event(device_data)
+                else:
                     LOG.debug("No changes in poll interval")
-                    continue
-                self._event(device_data)
-                time.sleep(1)
-            except requests.RequestException:
-                LOG.info("Could not contact Vera - will retry in %ss",
-                         SUBSCRIPTION_RETRY)
-                time.sleep(SUBSCRIPTION_RETRY)
-            except json.decoder.JSONDecodeError:
-                LOG.info("Response was malfomed - will retry in %ss",
-                         SUBSCRIPTION_RETRY)
-                time.sleep(SUBSCRIPTION_RETRY)
+
+                continue
```

Comment:
I think with you're restructuring this is now not doing anything - so can be removed.

Reply:
Check again. If you remove the continue, the flow will leave the try block and go into the "Could not contact Vera" path of the code that follows the try block--I don't think we want that.

Reply:
👍 You're right, sorry.